lib/rspec/mocks/targets.rb

```diff
@@ -38,7 +38,9 @@ def self.disallow_negation(method_name)
     private
 
       def matcher_allowed?(matcher)
-        matcher.class.name.start_with?("RSpec::Mocks::Matchers".freeze)
+        ancestors = matcher.class.ancestors
+        namespace = "RSpec::Mocks::Matchers"
+        ancestors.any? { |klass| klass.name.to_s.start_with? namespace }
```

Comment:
Amazing technics :)

It seems offhand, that if we go this way, it can lead us to extraction of the "essential/mockable" part of the matcher to the module. For now this is just a speculation, I need time to play with the idea.